src/tensors/abstracttensor.jl

```diff
@@ -480,6 +625,10 @@ end
 
 # Conversion to Array:
 #----------------------
+Base.ndims(t::AbstractTensorMap) = numind(t)
+Base.size(t::AbstractTensorMap) = ntuple(Base.Fix1(size, t), numind(t))
+Base.size(t::AbstractTensorMap, i) = dim(space(t, i))
```

Comment:
I am still not convinced about the usefulness of this. This is mostly aimed at making it more friendly for newcomers? But shouldn't we stimulate them to just learn from the start that `TensorMap`s are not just `AbstractArray`s and that they should just check the documentation to learn about the interface. I am happy to for example have a summary of the interface in the README or whathever can help.

One reason why I think this might not be a good idea is that this forces `length(t) = prod(size(t)) = dim(codomain) * dim(domain)`. But there might be an argument to be made for `length(t) = dim(codomain ← domain) = dim(t)`. So once we establish this is part of our public interface, it will be very hard to revert this if we ever come up with a better use case for these definitions.

Reply:
Jokes aside, I would actually be quite happy with a `TensorKit.allowscalar` to allow scalar `getindex`/`setindex` for abelian and `getindex` for non-abelian tensors too 😁